src/error.rs

```diff
@@ -1,60 +1,63 @@
 use std::error::Error as StdError;
 use std::fmt;
 
-/// The different types of errors that can occur while manipulating control groups.
-#[derive(Debug, Eq, PartialEq)]
-pub enum ErrorKind {
-    /// An error occured while writing to a control group file.
-    WriteFailed,
+/// The result type returned from this crate.
+pub type Result<T> = std::result::Result<T, Error>;
+
+/// The error type that can be returned from this crate, in the `Result::Err` variant.
+/// The lower-level cause of this error can be obtained from the `source` method.
+#[derive(Debug)]
+pub struct Error {
+    kind: ErrorKind,
+    source: Option<Box<dyn StdError + Send + 'static>>,
+}
 
-    /// An error occured while trying to read from a control group file.
+/// The kinds of errors that can occur while operating on cgroups.
+#[derive(Copy, Clone, Eq, PartialEq, Debug)]
+pub enum ErrorKind {
+    /// Failed to read from a cgroup file.
     ReadFailed,
 
-    /// An error occured while trying to parse a value from a control group file.
-    ///
-    /// In the future, there will be some information attached to this field.
-    ParseError,
+    /// Failed to parse a string in a cgroup file into a value.
+    /// In the future, there will be some information attached to this variant.
+    ParseFailed,
+
+    /// Failed to write to a cgroup file.
+    WriteFailed,
```

Comment:
Is there any reason for distinguishing error kinds for read and write operations?
IMO it is sufficient to have a single variant that indicates some IO operation was failed.